proto/profile.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package io.linkerd.proxy.profile;
+
+option go_package = "github.com/linkerd/linkerd2-proxy-api/go/profile";
+
+import "google/protobuf/duration.proto";
+
+import "http_types.proto";
+import "destination.proto";
+
+service Profile {
+  // Given a destination, return that service's profile and send an update
+  // whenever it changes.
+  rpc Get(destination.GetDestination) returns (stream ServiceProfile) {}
+}
+
+message ServiceProfile {
+  // A list of request polcies, each with a RequestMatch.  If a request matches
+  // more than one RequestPolicy, the first match wins.
+  repeated RequestPolicy request_policies = 1;
+  RetryBudget retry_budget = 2;
+  google.protobuf.Duration default_retry_timeout = 3;
+}
+
+message RetryBudget {
+  // The maximum ratio of retries to non-retry requests.  For example if this
+  // value is 0.2 then the number of retries is at most 20% of the number of
+  // non-retry requests.  This can be thought of as the amount of extra load
+  // added by retries (e.g. 20% extra load).
+  float max_retry_ratio = 1;
+}
+
+message RequestPolicy {
+  // This policy applies to requests that match this condition.
+  RequestMatch condition = 1;
+  // Responses that match this condition will be considered successful.  All
+  // other responses will be considered failures.
+  ResponseMatch success_condition = 2;
+  // Failure responses that match this condition may be retried.
+  ResponseMatch retry_condition = 3;
```

Comment:
I think we need to be able to enumerate and describe failure and retry conditions.  Success is the only condition that _doesn't_ warrant further description.